### PR TITLE
[DC-1643] ValidDeathDates should remove death_dates occurring before any PPI response

### PR DESCRIPTION
**Pull request by jp3477**
excluded records that have death before any ppi records.

### CODE REVIEW

File: data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py [hunk 1]
```diff
@@ -35,9 +35,24 @@
...
 SANDBOX_INVALID_DEATH_DATE_ROWS = common.JINJA_ENV.from_string("""
 CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{sandbox_table}}` AS (
-SELECT *
-FROM `{{project_id}}.{{dataset_id}}.{{table}}`
-WHERE {{table}}_date < '{{program_start_date}}' OR {{table}}_date > {{current_date}})
+SELECT d.*
+FROM `{{project_id}}.{{dataset_id}}.{{table}}` d
+LEFT JOIN (
+        SELECT
+            person_id, MIN(o.observation_date) first_ppi_date
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py:42]: This looks like it will only find the earliest date.  If I interpret 1643 correctly, we want to sandbox any death records that exist before any PPI records.  Added the following test code to check the current implementation when the death date falls between PPI records.  It fails with the existing implementation.
[1643_patch.txt](https://github.com/all-of-us/curation/files/6580336/1643_patch.txt)

File: data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py [hunk 2]
```diff
@@ -35,9 +35,24 @@
 # or after the current date.
 SANDBOX_INVALID_DEATH_DATE_ROWS = common.JINJA_ENV.from_string("""
 CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{sandbox_table}}` AS (
-SELECT *
-FROM `{{project_id}}.{{dataset_id}}.{{table}}`
-WHERE {{table}}_date < '{{program_start_date}}' OR {{table}}_date > {{current_date}})
+SELECT d.*
+FROM `{{project_id}}.{{dataset_id}}.{{table}}` d
+LEFT JOIN (
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py:40]: This is a really nice query.  Consider using Jinja `-- comment --` format to describe in plain English what this inner join is accomplishing.

**Review by lrwb-aou — CHANGES_REQUESTED** [review]
The updated test code fails the scenario where we have PPI record, death record, another PPI record.

File: data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py (hunk 1 above)
Comment (jp3477) [reply data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py:42]: Sounds good, will make that change.

File: data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py (hunk 2 above)
Comment (jp3477) [reply data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py:40]: Will add that, thanks

File: data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py (hunk 1 above)
Comment (jp3477) [reply data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py]: Made the change. I think just flipping the MIN to a MAX did the trick.

File: tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/valid_death_dates_test.py [hunk 3]
```diff
@@ -89,8 +119,16 @@ def setUp(self):
         fq_dataset_name = self.fq_table_names[0].split('.')
         self.fq_dataset_name = '.'.join(fq_dataset_name[:-1])
 
+        self.copy_vocab_tables()
+
         super().setUp()
 
+    def copy_vocab_tables(self):
```
Comment (lrwb-aou) [inline tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/valid_death_dates_test.py:126]: nit:  This is an optional thing, but I'm curious about why you are creating a function to run a loop instead of just running the loop.

Comment (jp3477) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/valid_death_dates_test.py:126]: I had copied this method from Nishanth, so you would have to ask him. But if I had to guess, it would be because it's more portable.

Comment (lrwb-aou) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/valid_death_dates_test.py:126]: I would suggest that this isn't portable because you're not importing this function and using it, you're redefining it.  I would recommend just using the loop and adding a comment above it about what it's doing.  This is adding a bit of extra complexity for just 2 lines of logical code.

Comment (jp3477) [reply tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/valid_death_dates_test.py:126]: Sounds good, will do that

**Review by lrwb-aou — APPROVED** [review]
The integration test here passed.  🎉   This is approved when the test cycle finishes.
